src/gui/attributeformconfig/qgsattributewidgetedit.cpp

```diff
@@ -115,7 +116,30 @@ void QgsAttributeWidgetRelationEditWidget::setRelationEditorConfiguration( const
       setCardinalityCombo( QStringLiteral( "%1 (%2)" ).arg( nmrel.referencedLayer()->name(), nmrel.fieldPairs().at( 0 ).referencedField() ), nmrel.id() );
   }
 
-  mRelationCardinalityCombo->setToolTip( tr( "For a many to many (N:M) relation, the direct link has to be selected. The in-between table will be hidden." ) );
+  int widgetTypeIdx = mWidgetTypeComboBox->findText( config.mRelationWidgetType );
+  mWidgetTypeComboBox->setCurrentIndex( widgetTypeIdx >= 0 ? widgetTypeIdx : 0 );
+
+  const QString widgetType = mWidgetTypeComboBox->currentData().toString();
+  mConfigWidget = QgsGui::relationWidgetRegistry()->createConfigWidget( widgetType, relation, this );
+  mConfigWidget->setConfig( config.mRelationWidgetConfig );
+  mWidgetTypePlaceholderLayout->addWidget( mConfigWidget );
+
+  disconnect( mWidgetTypeComboBoxConnection );
+
+  mWidgetTypeComboBoxConnection = connect( mWidgetTypeComboBox, &QComboBox::currentTextChanged, this, [ = ]()
+  {
+    const QString widgetId = mWidgetTypeComboBox->currentData().toString();
+
+    mConfigWidget->deleteLater();
+    mConfigWidget = QgsGui::relationWidgetRegistry()->createConfigWidget( widgetId, relation, this );
+    mConfigWidget->setConfig( config.mRelationWidgetConfig );
+    mWidgetTypePlaceholderLayout->addWidget( mConfigWidget );
+    update();
+  } );
+
+  mRelationCardinalityCombo->setToolTip( tr( "This is being changed" ) );
```

Comment:
What does this mean?

Reply:
Nothing, I have changed it while debugging and forgot to restore it. Thanks for spotting this.